cg_openmm/simulation/physical_validation.py

```diff
@@ -0,0 +1,282 @@
+            ensemble='NVT',
+            energy=state_energies[state2_index,:],
+            temperature=T_array[state2_index]
+            )
+            
+        sim_data2.units = sim_data1.units
+        
+        # Run physical validation
+        quantiles = pv.ensemble.check(
+            sim_data1,
+            sim_data2,
+            total_energy=False,
+            filename=plotfile
+            )
+        
+    elif pairs.lower() == 'adjacent':
+    # Run ensemble validation on all adjacent temperature pairs
+        quantiles = []
+
+        for i in range(len(temperature_list)-1):
+            # Set SimulationData for physical validation
+
+            # State 1
+            state1_index = i
+
+            sim_data1 = SimulationData()
+
+            sim_data1.observables = ObservableData(
+                potential_energy=state_energies[state1_index,:],
+                )
+                
+            sim_data1.ensemble = EnsembleData(
+                ensemble='NVT',
+                energy=state_energies[state1_index,:],
+                temperature=T_array[state1_index]
+                )
+                
+            sim_data1.units = UnitData(
+                kb=kB.value_in_unit(unit.kilojoule_per_mole/T_unit),
+                energy_conversion=1,
+                length_conversion=1,
+                volume_conversion=1,
+                temperature_conversion=1,
+                pressure_conversion=1,
+                time_conversion=1,
+                energy_str='KJ/mol',
+                length_str='nm',
+                volume_str='nm^3',
+                temperature_str='K',
+                pressure_str='bar',
+                time_str='ps'
+                )
+                
+            # State 2
+            state2_index = i+1
+            sim_data2 = SimulationData()
+
+            sim_data2.observables = ObservableData(
+                potential_energy=state_energies[state2_index,:],
+                )
+                
+            sim_data2.ensemble = EnsembleData(
+                ensemble='NVT',
+                energy=state_energies[state2_index,:],
+                temperature=T_array[state2_index]
+                )
+                
+            sim_data2.units = sim_data1.units
+            
+            # Run physical validation
+            quantiles_ij = pv.ensemble.check(
+                sim_data1,
+                sim_data2,
+                total_energy=False,
+                filename=f"{plotfile}_{state1_index}_{state2_index}"
+                )
+                
+            quantiles.append(quantiles_ij[0])
+    
+    elif pairs.lower() == 'all':
+    # Run ensemble validation on all combinations of temperature pairs
+        quantiles = []
+        for i in range(len(temperature_list)):
+            for j in range(i+1,len(temperature_list)):
+                # State 1
+                state1_index = i
+
+                sim_data1 = SimulationData()

```

Comment:
Could you write a single function that is used both in the single one, and loop over here, to reduce duplication?

Reply:
Added an internal function to set the SimulationData given the state_energies, temperature array, and state indices. How's this look? Seems to be producing the same results as before for 'single' and 'adjacent' options, 'all' still failing due to the bin overlap error - I can make a separate issue for that.

Reply:
> Added an internal function to set the SimulationData given the state_energies, temperature array, and state indices.

That change makes a lot more sense.

> Seems to be producing the same results as before for 'single' and 'adjacent' options, 'all' still failing due to the bin overlap error - I can make a separate issue for that.

That sounds fine.  I would see if you can catch the error using try/except syntax for each state.